V3 UI - Tabs & Menu rework

pythongosssss opened the pull request:
- Moves workflow tabs to the top level
  - Shrinks tabs down to a minumum size before overflowing
  - Add scroll arrows to make it more obvious when tabs dont fit
- Moves the app menu into the ComfyUI logo
  - On Desktop, right click context menu is the "native" context menu
- Moves the subgraph breadcrumbs next to the logo and always shows top level graph item
  - Progressively collapses the text/items at smaller window sizes
  - Add item specific commands to active workflow/subgraph (rename/duplicate/delete/etc)
- Optimized for Desktop with custom window chrome (Only Windows tested)
- Move "toggle bottom panel" button to left menu
- Move theme & settings to app menu
- Fix "clear" command deleting top level workflow when inside a subgraph
- Overflow scroll extension buttons

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-4374-V3-UI-Tabs-Menu-rework-2296d73d3650816db2facf7b12bc6421) by [Unito](https://www.unito.io)

![image](https://github.com/user-attachments/assets/9b5bc4d1-8630-4327-b09f-faa2e77a3612)

pythongosssss committed:
v3 interface updates
- move tabs to top
- improve tab resizing/scrolling
- rework menu
- reogranize top/side buttons

pythongosssss committed:
Refactor subgraph breadcrumbs
- Move to command bar
- Add dynamic overflow collapsing

pythongosssss committed:
Add breadcrumb menu

pythongosssss committed:
- desktop fixes
- breadcrumb collapse tweaks
- hot reload element observer fixes

pythongosssss committed:
- extension button overflow

pythongosssss committed:
- support custom extension actions
- fix some tests

pythongosssss committed:
Fix test due to removed mobile specific menu

pythongosssss committed:
Improve scrollbar styling on chrome

pythongosssss committed:
Fix test as menu is now not visible until clicked

JorgeR81 commented:
Looks really good !

I think it could replace some of my suggestions:

- https://github.com/Comfy-Org/ComfyUI_frontend/issues/1116
- https://github.com/Comfy-Org/ComfyUI_frontend/issues/3380
- https://github.com/Comfy-Org/ComfyUI_frontend/issues/3428
- https://github.com/Comfy-Org/ComfyUI_frontend/issues/4204

pythongosssss committed:
Fix test - change to use in memory screenshots to make less prone to breaking in future

pythongosssss committed:
- remove house icon on breadcrumbs
- increase size of logo/menu area with normal sidebar, add right spacing
- fix chevron icon inconsistency

pythongosssss committed:
- move tooltip style to more specific location
- add renaming by double clicking breadcrumb
- tweak submenu positions
- show menu items on hover (hack)

pythongosssss commented on src/assets/css/style.css:
```diff
@@ -637,3 +637,8 @@ audio.comfy-audio.empty-audio-widget {
+/* Fix tooltip going under menu, primevue config doesnt seem to work to change this */
+.p-tooltip {
+  z-index: 1100 !important;
+}
```
Done

pythongosssss committed:
Fix tests with menu changed to show on hover

Commit:
Update locales [skip ci]

pythongosssss committed:
Fix test

pythongosssss committed:
Merge branch 'pysssss/v3-interface-menu' of https://github.com/Comfy-Org/ComfyUI_frontend into pysssss/v3-interface-menu

christian-byrne reviewed (commented):
These are all changes which don't have to block merging, but we could consider adjusting in follow-up PRs:

## High Priority

1. **[src/components/breadcrumb/SubgraphBreadcrumb.vue]** - **Issue**: Brittle overflow calculation with hardcoded values. **Suggestion**: Explore CSS-based solutions or dynamic constant derivation.

2. **[src/components/breadcrumb/SubgraphBreadcrumbItem.vue]** - **Issue**: Direct DOM manipulation for width. **Suggestion**: Use reactive Vue styling.

3. **[src/components/topbar/CommandMenubar.vue:233-239]** - **Issue**: Direct manipulation of undocumented PrimeVue internal property (`.dirty`). **Suggestion**: Avoid internal hacks; find public API or implement custom logic. **HIGH PRIORITY FIX.**

4. **[src/components/topbar/CommandMenubar.vue]** - **Issue**: Use of `!important` in CSS. **Suggestion**: Prefer more specific selectors or component props.

5. **[src/components/topbar/WorkflowTabs.vue]** - **Issue**: Inconsistent horizontal scroll amount calculation. **Suggestion**: Refine `scrollAmount` for consistent behavior across devices.

6. **[src/components/breadcrumb/SubgraphBreadcrumbItem.vue:35]** - **Issue**: Extreme z-index (10000). **Suggestion**: Use design system value (1100-1200).

## Medium Priority

9. **[src/components/breadcrumb/SubgraphBreadcrumb.vue:58-64]** - **Issue**: Unsafe type casting for DOM access. **Suggestion**: Use proper TypeScript typing with null checks.

10. **[src/composables/element/useOverflowObserver.ts:34-37]** - **Issue**: Only checks horizontal overflow. **Suggestion**: Add option for vertical overflow detection.

11. **[src/components/topbar/WorkflowTabs.vue:220]** - **Issue**: Hardcoded scroll amount (20px). **Suggestion**: Make configurable or viewport-based.

12. **[src/components/topbar/WorkflowTabs.vue:376-386]** - **Issue**: Complex CSS with magic numbers and experimental env(). **Suggestion**: Add fallback styles and document compatibility.

13. **[src/components/topbar/CommandMenubar.vue:114-125]** - **Issue**: Recursive function without depth limit. **Suggestion**: Add maximum recursion depth protection.

## Low Priority

14. **[src/utils/mouseDownUtil.ts:3-7]** - **Issue**: No input validation. **Suggestion**: Add type guards and parameter validation.

15. **[src/components/breadcrumb/SubgraphBreadcrumbItem.vue:162-167]** - **Issue**: Relying on event.detail for click detection. **Suggestion**: Use more robust double-click handling.

16. **[src/components/breadcrumb/SubgraphBreadcrumb.vue:133-151]** - **Issue**: Complex manual width calculation. **Suggestion**: Use ResizeObserver or getBoundingClientRect().

christian-byrne reviewed (approved):
LGTM
